modules/search-service-impl/src/main/java/org/opencastproject/search/impl/persistence/SearchServiceDatabaseImpl.java

```diff
@@ -172,12 +173,14 @@ public void deleteMediaPackage(String mediaPackageId, Date deletionDate) throws
         String accessControlXml = searchEntity.get().getAccessControl();
 
         // allow ca users to retract live publications without putting them into the ACL
-        if (!(searchMp.isLive() && currentUser.hasRole(GLOBAL_CAPTURE_AGENT_ROLE))
-            && accessControlXml != null) {
-          AccessControlList acl = AccessControlParser.parseAcl(accessControlXml);
-          if (!AccessControlUtil.isAuthorized(acl, currentUser, currentOrg, WRITE.toString())) {
-            throw new UnauthorizedException(currentUser + " is not authorized to delete media package "
-                + mediaPackageId);
+        User user = securityService.getUser();
+        if (!user.hasRole(getEpisodeRoleId(mediaPackageId, WRITE.toString()))) {
+          if (!(searchMp.isLive() && currentUser.hasRole(GLOBAL_CAPTURE_AGENT_ROLE)) && accessControlXml != null) {
+            AccessControlList acl = AccessControlParser.parseAcl(accessControlXml);
+            if (!AccessControlUtil.isAuthorized(acl, currentUser, currentOrg, WRITE.toString())) {
+              throw new UnauthorizedException(
+                  currentUser + " is not authorized to delete media package " + mediaPackageId);
+            }
```

Comment:
This check and the ones below don't seem to check the config value that controls whether these special roles are activated?

Also: can this extra logic not move into `AccessControlUtil.isAuthorized`? Seems like thats the perfect place for it and would make all changes in this file unnecessary I think.

Reply:
Good point, we could also do that. ([ff61800](https://github.com/opencast/opencast/pull/5056/commits/ff61800f8c2014558062e9dceb256ca0d8a73314))